options/configurable.cc

```diff
@@ -19,6 +19,11 @@ namespace ROCKSDB_NAMESPACE {
 
 Configurable::Configurable() : is_prepared_(false) {}
 
+Configurable::Configurable(const Configurable& o)
+    : is_prepared_(o.is_prepared_.load()) {
+  options_ = o.options_;
+}
+
```

Comment:
I'd say let's remove `is_prepared_` / `IsPrepared` altogether. Two reasons:

1) It's unused except for unit tests.
2) This is not thread-safe regardless of whether the flag is atomic or not because the interface is inherently thread-*un*safe. I.e. in the presence of multiple threads, there is a race condition in the following even if you make the flag atomic because multiple threads may find the flag unset and proceed to call `PrepareOptions`:

```
if (!cfgable-IsPrepared()) {
   Status s = cfgable->PrepareOptions(cfg_options);
   // ...
}
```